### ISSUE TITLE
Deployment endpoint was not found

### ISSUE THREAD

**Issue by Stegosource** [GoogleCloudPlatform/github-actions#135]
### TL;DR
Running the `gcloud run deploy` command in GitHub Actions fails with the same command that succeeds locally. The error I keep seeing is `ERROR: (gcloud.run.deploy) Deployment endpoint was not found.`

**Expected behavior**
Deploy should work the same as when run locally

**Observed behavior**
When running locally, there are no problems. But when run from within GitHub Actions, I keep getting an error `ERROR: (gcloud.run.deploy) Deployment endpoint was not found.`
```
Run gcloud run deploy "***" \
  gcloud run deploy "***" \
    --quiet \
    --region "$RUN_REGION" \
    --image "gcr.io/***/***:latest" \
    --platform "managed" \
    --allow-unauthenticated

  shell: /bin/bash -e {0}
  env:
    PROJECT_ID: ***
    REPO_NAME: ***
    RUN_REGION: us-central1
    CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
    GOOGLE_APPLICATION_CREDENTIALS: ***
Deploying container to Cloud Run service [***] in project [***] region [us-central1]
Deploying new service...
failed
Deployment failed
ERROR: (gcloud.run.deploy) Deployment endpoint was not found.
##[error]Process completed with exit code 1.
```

### Reproduction
```yaml
# Docs: https://docs.github.com/en/actions
# Reference: https://docs.github.com/en/actions/reference
# Syntax: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions

name: CI

# Controls when the action will run.
# https://docs.github.com/en/actions/reference/events-that-trigger-workflows
on:
  push:
    branches: [ production, staging ]

env:
  PROJECT_ID: ${{ *** }}
  REPO_NAME: ${{ github.event.repository.name }}
  RUN_REGION: us-central1
  
jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2
        
#       https://github.com/GoogleCloudPlatform/github-actions/blob/master/setup-gcloud/README.md
      - name: Authenticate into Google Cloud Platform
        uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
        with:
          service_account_key: ${{ *** }} # base64 encoded keyfile.json contents
          project_id: ${{ env.PROJECT_ID }}
          export_default_credentials: true
          
      - name: Configure Docker to use Google Cloud Platform
        run: gcloud auth configure-docker --quiet
        
      - name: Build Docker image
        run: docker build . --tag "gcr.io/$PROJECT_ID/$REPO_NAME"
        
      - name: Push image to Google Cloud Container Registry
        run: docker push "gcr.io/$PROJECT_ID/$REPO_NAME"

      - name: Deploy to production
        if: github.ref == 'refs/heads/production' 
        run: |
          gcloud run deploy "$PROJECT_ID" \
            --quiet \
            --region "$RUN_REGION" \
            --image "gcr.io/$PROJECT_ID/$REPO_NAME" \
            --platform "managed" \
            --allow-unauthenticated
```

**Additional information**
Ive also tried hard coding the secrets and env variables, but still no luck.

**Comment by bharathkkb** [comment]
Hi @Stegosource. Thanks for opening this issue, this does seem odd, especially since you are passing in `--platform "managed" `. Nothing seems wrong off at a glance.  Could you set the [verbosity flag](https://cloud.google.com/sdk/gcloud/reference#--verbosity) to debug to capture some more info. @averikitsch any thoughts?

**Comment by sethvargo** [comment]
FWIW, I _think_ this:

```yaml
branches: [ production, staging ]
```

needs to be 

```yaml
branches: [ 'production', 'staging' ]
```

Tracing the code for gcloud, that error corresponds to a 404 when talking to Cloud Run. The most likely cause is a bad project ID or bad region. The region is correct, so please ensure you are using the **project ID**, not the project _name_ or project _number_. 

A more nuanced option is that your organization has restrictions (e.g. VPC-SC or network ACLs) that prohibit the communication to Cloud Run.

**Comment by Stegosource** [comment]
@bharathkkb and @sethvargo Thanks for looking. I think this was a user error issue from a bad copy/paste job. It's working now.

Re: the `production` vs `'production'` I dont think it matters. The docs seem to show it both ways, and the workflow is running. I'll keep an eye on it though in case some unexected behavior crops up. Thanks.
